libs/sdk-common/src/invoice.rs

```diff
@@ -101,6 +101,50 @@ fn format_short_channel_id(id: u64) -> String {
     format!("{block_num}x{tx_num}x{tx_out}")
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum Amount {
+    Bitcoin {
+        amount_msat: u64,
+    },
+    Currency {
+        // Reference to [FiatCurrency.id]
+        iso4217_code: String,
```

Comment:
Maybe a name that more obviously links it to FiatCurrency? something like `currency_id`

Reply:
This doesn't necessarily link it to `FiatCurrency`. It just happens they both use the same type of ID. `currency_id` might be obvious for us, but maybe not for others who aren't aware of `FiatCurrency`.

 IMO `iso4217_code` is clearer to the caller.